schema/schemaspec/extension.go

```diff
@@ -388,24 +426,27 @@ func specFields(ext interface{}) []fieldTag {
 		}
 		parts := strings.Split(lookup, ",")
 		fields = append(fields, fieldTag{
-			field:  f.Name,
-			tag:    lookup,
-			isName: len(parts) > 1 && parts[1] == "name",
+			field:            f.Name,
+			tag:              lookup,
+			isName:           len(parts) > 1 && parts[1] == "name",
+			isInterfaceSlice: f.Type.Kind() == reflect.Slice && f.Type.Elem().Kind() == reflect.Interface,
```

Comment:
What if it's not a slice? An array or just an interface?

Reply:
added support for just an interface. will support array in a subsequent diff